tls/s2n_config.c

```diff
@@ -532,6 +532,8 @@ static int s2n_config_add_cert_chain_and_key_impl(struct s2n_config *config, str
     POSIX_ENSURE_REF(config->domain_name_to_cert_map);
     POSIX_ENSURE_REF(cert_key_pair);
 
+    POSIX_GUARD_RESULT(s2n_security_policy_validate_certificate_chain(config->security_policy, cert_key_pair));
+
```

Comment:
One potential issue here would be operation ordering (again :( ). If a customer sets their cert chains before they set their security policies, you'll end up enforcing the default security policies here. That shouldn't be an issue since the default policies don't currently enable local enforcement, but that could change in the future. Minimally you probably need a test + comment warning.

Reply:
By test I'm thinking something that will fail if we decide to enable local enforcement on the default policies in the future.